File: utils/chatgpt_summarizing.py
import openai
import logging


def configure_openai(api_key: str, organization: str):
    openai.api_key = api_key
    openai.organization = organization


def summarize_text(text: str) -> str:
    input_text = f"Give me the main ideas of this tech article: {text[:4000]}"  # model limit
    model = "gpt-3.5-turbo"
    max_tokens = 500

    try:
        # Generate a summary using OpenAI's complete() function
        response = openai.ChatCompletion.create(
            model=model,
            messages=[
                {'role': 'user', 'content': input_text}
            ],
            temperature=0,
            max_tokens=max_tokens
        )

        # Extract the generated summary from the response
        summary = response['choices'][0]['message']['content'].strip()

        return summary
    except Exception as ex:
        logging.warning("Couldn't get summary for the text", ex)
        return ""

File: gorefactoring/utils/chatgptSummarizing.go
package modules

import (
	"context"
	"log"
	"strings"

	"github.com/sashabaranov/go-openai"
)

func ConfigureOpenAI(apiKey string) *openai.Client {
    return openai.NewClient(apiKey)
}

func SummarizeText(text string, client *openai.Client) string {
    modelLimit := 4000
    if len(text) > modelLimit {
        text = text[:4000]
    }
	inputText := "Give me the main ideas of this tech article: " + text
	model := openai.GPT4
	maxTokens := 300

	request := openai.ChatCompletionRequest{
		Model:      model,
        Messages: []openai.ChatCompletionMessage{
            {
                Role:    openai.ChatMessageRoleUser,
                Content:  inputText,
            },
        },
		Temperature: 0,
		MaxTokens:   maxTokens,
	}

	resp, err := client.CreateChatCompletion(
        context.Background(),
        request,
    )
	if err != nil {
		log.Println("Couldn't get summary for the text:", err)
		return ""
	}

	summary := resp.Choices[0].Message.Content
	return strings.TrimSpace(summary)
}